src/regression.js

```diff
@@ -30,12 +30,26 @@
   /**
    * Determine the coefficient of determination (r^2) of a fit from the observations and predictions.
    *
+   * @param {Array<Array<number>>} data - Pairs of observed x-y values, including nulls
+   * @param {Array<Array<number>>} results - Pairs of observed predicted x-y values, including null predictions
+   *
    * @param {Array<Array<number>>} observations - Pairs of observed x-y values
    * @param {Array<Array<number>>} predictions - Pairs of observed predicted x-y values
    *
    * @return {number} - The r^2 value, or NaN if one cannot be calculated.
    */
-  function determinationCoefficient(observations, predictions) {
+  function determinationCoefficient(data, results) {
+    var observations = [];
+    var predictions = [];
+
+    // Remove predictive points for accurate fit
+    data.forEach( function (d, i) {
```

Comment:
No, not exactly. If you look here: https://github.com/looker/regression-js/pull/1/files#diff-e923190f420baf3e60f5b9832f4dc71cR170
we've filtered out nulls for calculating the gradient and intercept, but then map the resulting equation across _all_ data, including points with a null y. So the resulting data takes every x point into account.